src/CTA.Rules.Actions/ActionHelpers/ConfigMigrate.cs

```diff
@@ -67,76 +71,105 @@ private XDocument LoadWebConfig(string projectDir)
         }
 
         /// <summary>
-        /// Deserialize web.config file to an object
+        /// Reads data from web.config and writes any relevant entries to appsettings.json file
         /// </summary>
-        /// <param name="projectDir">The project directory containing the web.config file</param>
-        /// <returns></returns>
-        private string ProcessWebConfig(XDocument webConfig)
+        /// <param name="webConfig">The XML representation of the web.config file</param>
+        /// <returns>A JSON object representing the appSettings.json file </returns>
+        private JObject ProcessWebConfig(Configuration webConfig, string templateContent)
         {
-            StringBuilder config = new StringBuilder();
+            var connectionStringsObjects = GetConnectionStrings(webConfig);
+            var appSettingsObjects = GetAppSettingObjects(webConfig);
 
-            webConfig.Descendants()?.Where(d => d.Name.ToString().ToLower() == Constants.connectionstrings)?.Descendants()?.ToList().ForEach(connectionString =>
+            if(_hasData)
             {
```

Comment:
The value of `_hasData` is set in multiple places- what do you think about setting that value here instead, using something like:
```
_hasData = connectionStringsObjects.Any() || appSettingsObjects.Any();
```
By setting its value in one place, it may make this easier to follow and readers will immediately know all conditions in which this flag is set to `true`.

Reply:
Ok